saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/store/db/AbstractStore.java

```diff
@@ -0,0 +1,221 @@
+ *  limitations under the License.
+ */
+package io.seata.saga.engine.store.db;
+
+import io.seata.common.util.ReflectionUtil;
+import io.seata.saga.engine.store.StoreException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.sql.DataSource;
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Abstract store
+ * @author lorne.cl
+ */
+public class AbstractStore {
+
+    protected static final Logger LOGGER = LoggerFactory.getLogger(AbstractStore.class);
+
+    protected DataSource dataSource;
+
+    protected String dbType;
+
+    protected String tablePrefix;
+
+    protected <T> T selectOne(String sql, ResultSetToObject<T> resultSetToObject, Object... args){
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
+            connection.setAutoCommit(true);
+
+            if(LOGGER.isDebugEnabled()){
+                LOGGER.debug("Preparing SQL statement: {}", sql);
+            }
+
+            PreparedStatement stmt = connection.prepareStatement(sql);
+
+            if(LOGGER.isDebugEnabled()){
+                LOGGER.debug("setting params to PreparedStatement: {}", Arrays.toString(args));
+            }
+
+            for(int i = 0; i < args.length; i++){
+                stmt.setObject(i+1, args[i]);
+            }
+            ResultSet resultSet = stmt.executeQuery();
+            if(resultSet.next()){
+                return resultSetToObject.toObject(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new StoreException(e);
+        } finally {
+            closeSilent(connection);
+        }
+        return null;
+    }
+
+    protected <T> List<T> selectList(String sql, ResultSetToObject<T> resultSetToObject, Object... args){
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
+            connection.setAutoCommit(true);
+
+            if(LOGGER.isDebugEnabled()){
+                LOGGER.debug("Preparing SQL: {}", sql);
+            }
+
+            PreparedStatement stmt = connection.prepareStatement(sql);
+
+            if(LOGGER.isDebugEnabled()){
+                LOGGER.debug("setting params to PreparedStatement: {}", Arrays.toString(args));
+            }
+
+            for(int i = 0; i < args.length; i++){
+                stmt.setObject((i+1), args[i]);
+            }
+            ResultSet resultSet = stmt.executeQuery();
+            List<T> list = new ArrayList<>();
+            while (resultSet.next()){
+                list.add(resultSetToObject.toObject(resultSet));
+            }
+            return list;
+        } catch (SQLException e) {
+            throw new StoreException(e);
+        } finally {
+            closeSilent(connection);
```

Comment:
It is recommended to manually close stmt, resultSet

Reply:
fixed